src/Marten/Events/Daemon/TenantActionGroup.cs

```diff
@@ -22,8 +22,14 @@ public TenantActionGroup(Tenant tenant, IEnumerable<StreamAction> actions)
     public async Task ApplyEvents(ProjectionUpdateBatch batch, IProjection projection, DocumentStore store,
         CancellationToken cancellationToken)
     {
+        var tracking =
```

Comment:
Not now, but I think the Daemon code gets some serious love soon. I'd prefer to be passing `AsyncOptions` around explicitly w/o the optional downcast. Do that when we tackle first class subscriptions. We'll use `AsyncOptions` with subscriptions I'd imagine.

Reply:
Yes, I wasn't fully happy about that, but I wanted to make it less invasive. I think that while working on Subscriptions, and especially on v7 we should revisit our abstractions on top of projections and maybe shuffle them a bit to make them more straightforward.